src/metadatautil.c

```diff
@@ -83,14 +96,25 @@ create_enum_input( VipsImage *image, const gchar *field, GParamSpec *pspec )
 		for( int i = 0; i < pspec_enum->enum_class->n_values; ++i )
 			nicks[i] = pspec_enum->enum_class->values[i].value_nick;
 		nicks[pspec_enum->enum_class->n_values] = NULL;
-		t = gtk_drop_down_new_from_strings( nicks );
+
 		g_object_get( image, field, &d, NULL );
+
+#ifdef EXPERIMENTAL_METADATA_EDIT
+		t = gtk_drop_down_new_from_strings( nicks );
 		gtk_drop_down_set_selected( GTK_DROP_DOWN( t ), d );
+#else
+		t = gtk_label_new( g_strdup_printf( "%s", nicks[d] ) );
```

Comment:
I don't think this will work -- `d` is the value of the enum, not the index of the enum in `nicks`.

How about:

```C
        t = gtk_label_new( vips_enum_nick( pspec->value_type, d ) );
```

Reply:
I pull `nicks` off the `GParamSpecEnum` (if it exists). The order of the elements in this array should match the order of the enum. I'll use vips_enum_nick anyway if possible.

```
...
                GParamSpecEnum *pspec_enum = G_PARAM_SPEC_ENUM( pspec );
                ...
                const char **nicks;

                nicks = g_malloc( (pspec_enum->enum_class->n_values + 1) * sizeof( char * ) );

                for( int i = 0; i < pspec_enum->enum_class->n_values; ++i )
                        nicks[i] = pspec_enum->enum_class->values[i].value_nick;
                nicks[pspec_enum->enum_class->n_values] = NULL;

...

```

Reply:
> ```c
> vips_enum_nick
> ```

Done